mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -241,11 +261,17 @@ class ALSModel private[ml] (
         Float.NaN
       }
     }
-    dataset
+    val predictions = dataset
       .join(userFactors, dataset($(userCol)) === userFactors("id"), "left")
       .join(itemFactors, dataset($(itemCol)) === itemFactors("id"), "left")
       .select(dataset("*"),
         predict(userFactors("features"), itemFactors("features")).as($(predictionCol)))
+    $(unknownStrategy) match {
+      case ALSModel.Drop =>
```

Comment:
Sure, I just think it's easier to maintain with changes to the options
occurring in one place, just for avoiding duplication. But it doesn't make
a big difference
On Wed, 4 May 2016 at 18:17, Seth Hendrickson <notifications@github.com>
wrote:

> In mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala
> <https://github.com/apache/spark/pull/12896#discussion_r62068526>:
>
> >        .join(userFactors, dataset($(userCol)) === userFactors("id"), "left")
> >        .join(itemFactors, dataset($(itemCol)) === itemFactors("id"), "left")
> >        .select(dataset("*"),
> >          predict(userFactors("features"), itemFactors("features")).as($(predictionCol)))
> > +    $(unknownStrategy) match {
> > +      case ALSModel.Drop =>
>
> For my own curiosity, is there a reason we can't just match on the string
> value and avoid creating these new vals? Seems like I have seen both ways
> in the code.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12896/files/fc437451a598221f0878b7a2e0b87d17572019cc#r62068526>
>